app/Models/Server.php

```diff
@@ -438,17 +438,17 @@ public function send(array|string $command): ResponseInterface
         ])->toPsrResponse();
     }
 
-    public function retrieveStatus(): string
+    public function retrieveStatus(): ContainerStatus
     {
         $status = cache()->get("servers.$this->uuid.container.status");
 
-        if ($status) {
-            return $status;
-        }
+        if ($status === null) {
+            $this->node->serverStatuses();
 
-        $this->node->serverStatuses();
+            $status = cache()->get("servers.$this->uuid.container.status");
```

Comment:
`$this->node->serverStatuses()` already returns the new statuses. You don't have to access the cache for it.